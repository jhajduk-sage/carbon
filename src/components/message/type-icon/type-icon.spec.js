import React from "react";
import TestRenderer from "react-test-renderer";
import TypeIconStyle from "./type-icon.style";

function render(props) {
  return TestRenderer.create(<TypeIconStyle {...props} />);
}

const messages = ["info", "error", "success", "warning"];

describe("TypeIcon", () => {
  describe("when rendered", () => {
    describe("with no additional props", () => {
      messages.forEach((variant) => {
        it(`should match the snapshot for ${variant}`, () => {
          const wrapper = render({ variant });
          expect(wrapper).toMatchSnapshot();
        });
      });
    });
  });

  describe("when transparent prop is set to true", () => {
    it("applies white background and the type icon with the proper style applied", () => {
      messages.forEach((variant) => {
        const wrapper = render({ transparent: true, variant });
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
