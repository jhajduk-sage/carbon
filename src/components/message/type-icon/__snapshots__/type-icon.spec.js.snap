// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeIcon when rendered with no additional props should match the snapshot for error 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #C7384F;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 100%;
  min-width: 30px;
  text-align: center;
}

.c0 span:before {
  color: #FFFFFF;
}

<div
  className="c0"
/>
`;

exports[`TypeIcon when rendered with no additional props should match the snapshot for info 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0073C2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 100%;
  min-width: 30px;
  text-align: center;
}

.c0 span:before {
  color: #FFFFFF;
}

<div
  className="c0"
/>
`;

exports[`TypeIcon when rendered with no additional props should match the snapshot for success 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #00B000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 100%;
  min-width: 30px;
  text-align: center;
}

.c0 span:before {
  color: #FFFFFF;
}

<div
  className="c0"
/>
`;

exports[`TypeIcon when rendered with no additional props should match the snapshot for warning 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #E96400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 100%;
  min-width: 30px;
  text-align: center;
}

.c0 span:before {
  color: #FFFFFF;
}

<div
  className="c0"
/>
`;

exports[`TypeIcon when transparent prop is set to true applies white background and the type icon with the proper style applied 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0073C2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 100%;
  min-width: 30px;
  text-align: center;
  background-color: transparent;
}

.c0 span:before {
  color: #FFFFFF;
}

.c0 span:before {
  color: #0073C2;
}

<div
  className="c0"
/>
`;

exports[`TypeIcon when transparent prop is set to true applies white background and the type icon with the proper style applied 2`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #C7384F;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 100%;
  min-width: 30px;
  text-align: center;
  background-color: transparent;
}

.c0 span:before {
  color: #FFFFFF;
}

.c0 span:before {
  color: #C7384F;
}

<div
  className="c0"
/>
`;

exports[`TypeIcon when transparent prop is set to true applies white background and the type icon with the proper style applied 3`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #00B000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 100%;
  min-width: 30px;
  text-align: center;
  background-color: transparent;
}

.c0 span:before {
  color: #FFFFFF;
}

.c0 span:before {
  color: #00B000;
}

<div
  className="c0"
/>
`;

exports[`TypeIcon when transparent prop is set to true applies white background and the type icon with the proper style applied 4`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #E96400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 100%;
  min-width: 30px;
  text-align: center;
  background-color: transparent;
}

.c0 span:before {
  color: #FFFFFF;
}

.c0 span:before {
  color: #E96400;
}

<div
  className="c0"
/>
`;
